### DISCUSSION TITLE
Deploying a custom spacy ner model

### DISCUSSION THREAD

**Discussion by einmdrd** [explosion/spaCy#13513] [Help: Other Questions]
Do you have some steps, tutorials or guides on how to deploy a custom made spacy ner model as api?

I used fastapi and spacy for my server, but I need to deploy it in order for my flutter mobile app to access it. I already created the model as a package so I can pip install it, but it failed to deploy. But everything works fine, when the server is run on the local machine and the app on emulator. How do you think should I set up my files to deploy it successfully?

**Comment by PiwanSama** [comment]
I'm not sure if you got this answered, but using FastAPI it would be something like:

@app.post("/ner/")
async def get_entities(request: str):
    try:
        doc = nlp(request)
        entities = [{"text": ent.text, "label": ent.label_} for ent in doc.ents]
        return {"entities": entities}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error processing request: {e}")

You can then deploy the entire service inside a Docker container

**Comment by igormorgado** [comment]
Here https://github.com/explosion/aiGrunn-2023/  in `annotation-service` you can see a example of a fastapi service using a model and returning annotations. To use a custom model instead to point to the common model names as `en_web_core_trf`, just point to the location of your trained model as `training/model-best`.  Or create a package with your model and use it.

Hope it helps.
